core/scripts/IntegrationTests.js

```diff
@@ -0,0 +1,344 @@
+
+    for (const account of accounts) {
+      // approve the tokens
+      await collateralToken.approve(expiringMultiParty.address, mintAndApprove, {
+        from: account
+      });
+      await syntheticToken.approve(expiringMultiParty.address, mintAndApprove, {
+        from: account
+      });
+
+      // mint collateral for all accounts
+      await collateralToken.mint(account, mintAndApprove, { from: contractCreator });
+    }
+  });
+  it("Iterative full life cycle test with friendly numbers", async function() {
+    /**
+     * @notice Iterative test with sponsors, liquidations and disputes.
+     * Number of positions to create and liquidate. The following process is followed to
+     * initiate maximum interaction with the emp & fee paying function to try and compound
+     *  floating errors to see if positions are locked at settlement:
+     * 0.a) set the oracle fee
+     * 0.b) create a large position by the liquidator to enable them to perform liquidations
+     * 1) position created by selected sponsor
+     * 2) random amount of tokens sent to a selected tokenholder
+     * 3) time advanced by 1000 seconds
+     * 4) 1/3 chance to initiate liquidation
+     * 4.a) if liquidation initiated then time advanced
+     * 4.b) 1/2chance to dispute
+     * 4.b.i) if disputed then resolve oracle price
+     * 5) chance for token sponsor to deposit more collateral
+     * 6) chance for the sponsor to redeem collateral
+     * 7) repeat 1 to 6 `numIterations` times
+     * 8) withdraw successful (or failed) liquidation returns from sponsors, liquidators and disputers
+     * 9) settle contract
+     * 10) ensure that all users can withdraw their funds
+     * 11) check the contract has no funds left in it
+     */
+
+    // Test settings
+    const numIterations = 20; // number of times the simulation loop is run
+    const runLiquidations = true; // if liquidations should occur in the loop
+    const runDisputes = true; // if disputes should occur in the loop
+    const runExtraDeposits = true; // if the sponsor should have a chance to add more
+    const runRedeemTokens = true; // if the sponsor should have a chance to redeem some of their tokens
+
+    // Tunable parameters
+    const baseCollateralAmount = toBN(toWei("150")); // starting amount of collateral deposited by sponsor
+    const baseNumTokens = toBN(toWei("100")); // starting number of tokens created by sponsor
+    const settlementPrice = toBN(toWei("1")); // Price the contract resolves to
+    const liquidationPrice = toBN(toWei("1.5")); // Price a liquidator will liquidate at
+    const disputePrice = toBN(toWei("1")); // Price a dispute will resolve to
+    const depositAmount = toBN(toWei("10")); // Amount of additional collateral to add to a position
+    const redeemAmount = toBN(toWei("1")); // The number of synthetic tokens to redeem for collateral
+    const dvmRegularFee = toBN("0"); // Fee multiplier
```

Comment:
Comments a bit misleading, its not the fee multiplier its the fee. Also the fee is a FixedPoint so you should probably `toWei` it. For example, if you want to charge 1% fees per second, you would set the fee to `toWei("0.01")`

Reply:
updated.